esp-open-rtos.patch

```diff
@@ -236,193 +32,17 @@ index 48b2f14..2062f24
-+	_mutex = xSemaphoreCreateMutex();
- 	if (tz) {
- 		stz = *tz;
- 	} else {
-@@ -51,36 +58,36 @@ void sntp_initialize(const struct timezone *tz) {
- 	sntp_base = 0;
- 	// To avoid div by 0 exceptions if requesting time before SNTP config
- 	cal = 1;
--	tim_ref = TIMER_COUNT;
-+	prev_timer_count = TIMER_COUNT;
- 	sntp_init();
- }
- 
--// Check if a timer wrap has occurred. Compensate sntp_base reference
--// if affirmative.
--// TODO: think about multitasking and race conditions
--static inline void sntp_check_timer_wrap(uint32_t current_value) {
--	if (current_value < tim_ref) {
--		// Timer wrap has occurred, compensate by subtracting 2^32 to ref.
--		sntp_base -= 1LLU<<32;
--		// DEBUG
--		printf("\nTIMER WRAPPED!\n");
--	}
--}
--
- // Return secs. If us is not a null pointer, fill it with usecs
- inline time_t sntp_get_rtc_time(int32_t *us) {
- 	time_t secs;
- 	uint32_t tim;
- 	uint64_t base;
-+	bool useMutex = _mutex != NULL;
- 
-+	if (useMutex && xSemaphoreTake(_mutex, 1000) != pdTRUE) {
-+		printf("%s SemTake failed\n", __FUNCTION__);
-+	}
- 	tim = TIMER_COUNT;
- 	// Check for timer wrap
--	sntp_check_timer_wrap(tim);
--	base = sntp_base + tim - tim_ref;
-+	if (tim < prev_timer_count) {
-+		// Timer wrap has occurred
-+		sntp_base += 1LLU<<32;
-+		printf("\nTIMER WRAPPED!\n");
-+	}
-+	prev_timer_count = tim;
-+	base = sntp_base + tim;
- 	secs = base * cal / (1000000U<<12);
- 	if (us) {
- 		*us = base * cal % (1000000U<<12);
- 	}
-+	if (useMutex && xSemaphoreGive(_mutex) != pdTRUE) {
-+		printf("%s SemTake failed\n", __FUNCTION__);
-+	}
- 	return secs;
- }
- 
-@@ -97,16 +104,26 @@ int _gettimeofday_r(struct _reent *r, struct timeval *tp, void *tzp) {
- 
- // Update RTC timer. Called by SNTP module each time it receives an update.
- void sntp_update_rtc(time_t t, uint32_t us) {
-+	if (xSemaphoreTake(_mutex, 1000) != pdTRUE) {
-+		printf("%s SemTake failed\n", __FUNCTION__);
-+	}
- 	// Apply daylight and timezone correction
- 	t += (stz.tz_minuteswest + stz.tz_dsttime * 60) * 60;
- 	// DEBUG: Compute and print drift
--	int64_t sntp_current = sntp_base + TIMER_COUNT - tim_ref;
-+	int64_t sntp_current = sntp_base + TIMER_COUNT;
- 	int64_t sntp_correct = (((uint64_t)us + (uint64_t)t * 1000000U)<<12) / cal;
--	printf("\nRTC Adjust: drift = %ld ticks, cal = %d\n", (time_t)(sntp_correct - sntp_current), cal);
- 
--	tim_ref = TIMER_COUNT;
-+	prev_timer_count = TIMER_COUNT;
- 	cal = sdk_system_rtc_clock_cali_proc();
- 
- 	sntp_base = (((uint64_t)us + (uint64_t)t * 1000000U)<<12) / cal;
-+	sntp_base -= prev_timer_count;
-+
-+	if (xSemaphoreGive(_mutex) != pdTRUE) {
-+		printf("%s SemTake failed\n", __FUNCTION__);
-+	}
-+	printf("base = 0x%x%x, , ref: %u\n",
-+			(uint32_t)(sntp_base >>32 & 0xffffffff), (uint32_t)(sntp_base & 0xffffffff), prev_timer_count );
-+	printf("RTC Adjust: drift = %ld ticks, cal = %d\n", (time_t)(sntp_correct - sntp_current), cal);
- }
- 
-diff --git a/include/ssid_config.h b/include/ssid_config.h
```

Comment:
Als ik het archief clone, de update_rtos branch uitcheck, de bouw instructies uit readme.md volg krijg ik onderstaande error: 
In file included from /home/robert/proj/esp8266_wordclock_new_rtos/wordclock/ota_basic.c:17:0:
../esp-open-rtos/include/ssid_config.h:16:33: fatal error: private_ssid_config.h: No such file or directory
 #include "private_ssid_config.h"